BibBuddy/app/src/main/java/de/bibbuddy/BookDao.java

```diff
@@ -88,19 +88,23 @@ public Book findById(Long id) {
     if (cursor != null) {
       cursor.moveToFirst();
 
-      book = new Book(
-          Long.parseLong(cursor.getString(0)), // Id
-          cursor.getString(1), // Isbn
-          cursor.getString(2), // Title
-          cursor.getString(3), // Subtitle
-          Integer.parseInt(cursor.getString(4)), // Pub year
-          cursor.getString(5), // Publisher
-          cursor.getString(6), // Volume
-          cursor.getString(7), // Edition
-          cursor.getString(8), // Add Infos
-          Integer.parseInt(cursor.getString(9)), // Create date
-          Integer.parseInt(cursor.getString(10)) // Mod date
-      );
+      book = new Book();
+      book.setId(Long.parseLong(cursor.getString(0))); // Id
+      book.setIsbn(cursor.getString(1));  // Isbn
+      book.setTitle(cursor.getString(2)); // Title
+      book.setSubtitle(cursor.getString(3)); // Subtitle
+
+      String pubYearStr = cursor.getString(4);
+      if (pubYearStr != null) {
+        book.setPubYear(Integer.parseInt(pubYearStr)); // Pub year
+      }
```

Comment:
Verification should be done at other place.

Reply:
In the database the publisher year can be null. And if you want to find a book by id you can't verify it before if it is null because you need to find it first. If you try to convert a null value to an int you get an error.